### PR TITLE
Skip zero length file in hdfs scan node

### LINKED ISSUE #4534: Hive external table uses OSS format to query and wait all the time
### Steps to reproduce the behavior (Required)
1. In hive environment
CREATE external TABLE hive_hdfs_decimal_oss_orc (
        col_int int comment "column int"
        ,col_decimal_01 DECIMAL(9,4) comment "column DECIMAL(9,4)"
        ,col_decimal_02 DECIMAL(18,9) comment "column DECIMAL(18,9)"
        ,col_decimal_03 DECIMAL(38,19) comment "column DECIMAL(38,19)"
) 
 stored as orc
location 'oss://dorisdb-data/hive_external_tbl_array_decimalV3/hive_hdfs_decimal_hdfs_orc';

CREATE external TABLE hive_hdfs_decimal_oss_csv (
        col_int int comment "column int"
        ,col_decimal_01 DECIMAL(9,4) comment "column DECIMAL(9,4)"
        ,col_decimal_02 DECIMAL(18,9) comment "column DECIMAL(18,9)"
        ,col_decimal_03 DECIMAL(38,19) comment "column DECIMAL(38,19)"
)
row format delimited fields terminated by ','
 stored as textfile
location 'oss://dorisdb-data/hive_external_tbl_array_decimalV3/hive_hdfs_decimal_hdfs_csv'

2. In Starrocks environment
create external table ex_hive_tbl_dec_oss (
                    col_int int comment "column int"
                    ,col_decimal_01 DECIMAL32(9,4) comment "column DECIMAL(9,4)"
                    ,col_decimal_02 DECIMAL64(18,9) comment "column DECIMAL(18,9)"
                    ,col_decimal_03 DECIMAL128(38,19) comment "column DECIMAL(38,19)"
       )
                ENGINE=hive
                properties (
                    "resource" = "hive_resource_hy",
                    "table" = "hive_hdfs_decimal_oss_orc",
                    "database" = "hive_test_hy");

create external table ex_hive_tbl_dec_oss_csv (
                    col_int int comment "column int"
                    ,col_decimal_01 DECIMAL32(9,4) comment "column DECIMAL(9,4)"
                    ,col_decimal_02 DECIMAL64(18,9) comment "column DECIMAL(18,9)"
                    ,col_decimal_03 DECIMAL128(38,19) comment "column DECIMAL(38,19)"
       )
                ENGINE=hive
                properties (
                    "resource" = "hive_resource_hy",
                    "table" = "hive_hdfs_decimal_oss_csv",
                    "database" = "hive_test_hy")

3. 
select * from ex_hive_tbl_dec_oss;
select * from ex_hive_tbl_dec_csv;

### Expected behavior (Required)

+---------+----------------+----------------+-----------------------+
| col_int | col_decimal_01 | col_decimal_02 | col_decimal_03        |
+---------+----------------+----------------+-----------------------+
|       1 |         1.1000 |    1.100000000 | 1.1000000000000000000 |
+---------+----------------+----------------+-----------------------+
1 row in set (0.52 sec)

### Real behavior (Required)
The query statement has no result and has been waiting

### StarRocks version (Required)
mysql> select current_version();
+------------------------+
| current_version()      |
+------------------------+
| QA_TEST_MASTER f6f6228 |
+------------------------+
1 row in set (0.01 sec)

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4534

### CODE REVIEW

File: be/src/exec/vectorized/hdfs_scan_node.cpp [hunk 1]
```diff
@@ -577,7 +577,9 @@ Status HdfsScanNode::close(RuntimeState* state) {
 
 Status HdfsScanNode::set_scan_ranges(const std::vector<TScanRangeParams>& scan_ranges) {
     for (const auto& scan_range : scan_ranges) {
-        _scan_ranges.emplace_back(scan_range.scan_range.hdfs_scan_range);
+        auto& range = scan_range.scan_range.hdfs_scan_range;
+        if (range.file_length == 0) continue;
```
Comment (imay) [inline be/src/exec/vectorized/hdfs_scan_node.cpp:581]: better to give some comment and why not eliminate these at Frontend.

Comment (dirtysalt) [reply be/src/exec/vectorized/hdfs_scan_node.cpp:581]: yes, better to remove at frontend. And for safety/backward-compatibility, we can also keep this line.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 4 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/external/RemoteScanRangeLocations.java|0|2|00.00%|

**Review by Youngwb — APPROVED** [review]

**Review by imay — APPROVED** [review]
LGTM

**Comment by dirtysalt** [comment]
@Mergify backport branch-2.2

**Pull request merged**